logics/impls/rmrk/multiasset.rs

```diff
@@ -0,0 +1,398 @@
+//! This module enables multiasset capability of RMRK
+
+use crate::impls::rmrk::{
+    errors::RmrkError,
+    types::*,
+};
+pub use crate::traits::multiasset::{
+    Internal,
+    MultiAsset,
+    MultiAssetEvents,
+};
+use ink_prelude::vec::Vec;
+use openbrush::{
+    contracts::{
+        ownable::*,
+        psp34::extensions::enumerable::*,
+    },
+    modifiers,
+    traits::{
+        AccountId,
+        Storage,
+        String,
+    },
+};
+
+/// Implement internal helper trait for MultiAsset
+impl<T> Internal for T
+where
+    T: Storage<MultiAssetData> + Storage<psp34::Data<enumerable::Balances>>,
+{
+    /// Check if token is minted. Return the token uri
+    fn asset_id_exists(&self, asset_id: AssetId) -> Option<String> {
+        if let Some(index) = self
+            .data::<MultiAssetData>()
+            .collection_asset_entries
+            .iter()
+            .position(|a| a.asset_id == asset_id)
+        {
+            let asset_uri =
+                &self.data::<MultiAssetData>().collection_asset_entries[index].asset_uri;
+            return Some(asset_uri.clone())
+        }
+
+        None
+    }
+
+    /// Check if token is minted. Return the owner
+    fn ensure_exists(&self, id: &Id) -> Result<AccountId, PSP34Error> {
+        let token_owner = self
+            .data::<psp34::Data<enumerable::Balances>>()
+            .owner_of(id.clone())
```

Comment:
```suggestion
            .owner_of(*id)
```
https://github.com/rust-lang/rust-clippy/issues/2870

Reply:
`Id` type does not implement copy() and cant't be dereferenced.